### DISCUSSION TITLE
how to remove legend?

### DISCUSSION THREAD

**Discussion by kernd** [pyqtgraph/pyqtgraph#1930] [Q&A]
how can I remove a legend from a PlotWidet? I tried the clear() method, but this does not clear the legend.

**Comment by ntjess** [comment]
Hi @kernd, do you mean to avoid the legend entirely or to remove it once it's added? If you don't manually add a legend, it should not be there by default:

```python
pw = pg.PlotWidget()
pw.show() # No legend present
```

If you added a legend, you can remove it using the scene's `removeItem`:

```python
import pyqtgraph as pg
pg.mkQApp()
pw = pg.PlotWidget()
legend = pw.addLegend()
scatter = pg.ScatterPlotItem([0, 1, 2], [0, 1, 2], name='test')
pw.addItem(scatter) # Legend is visible
pw.scene().removeItem(legend) # Legend is gone
pw.show()
```

**Comment by klim-** [comment]
Removing the item from the scene seems to work, however, in my application I want to be able to dynamically show and hide the legend (which can be dragged to other parts of the plot by mouse). Unfortunately, adding the item back to the scene is not sufficient, but results in errors as shown in the following example:

```python
import numpy as np
import pyqtgraph as pg

win = pg.plot()

# curve
c1 = win.plot([np.random.randint(0,8) for i in range(10)], pen='r', symbol='t', symbolPen='r', symbolBrush='g', name='curve1')
c2 = win.plot([2,1,4,3,1,3,2,4,3,2], pen='g', fillLevel=0, fillBrush=(255,255,255,30), name='curve2')

# legend
legend = pg.LegendItem((80,60), offset=(70,20))
legend.setParentItem(win.graphicsItem())
legend.addItem(c1, 'curve1')
legend.addItem(c2, 'curve2')

# this seems to work
win.scene().removeItem(legend)

# errors if mouse is used to drag legend to some other place after adding the legend back
win.scene().addItem(legend)

if __name__ == '__main__':
    pg.exec()
```

What am I missing here?

**EDIT:** It seems that calling `legend.setParentItem(win.graphicsItem())` again would do the trick, even though this would not remember the position if the legend has been moved via mouse drag. Any solution to that?

**EDIT 2:** The solution to the positioning problem seems to be to use `legend.pos()` instead of `legend.opts['offset']`. Perhaps this is helpful.
